docs/proposals/hpa/cronfederatedhpa.md

```diff
@@ -0,0 +1,339 @@
+1. Support both `FederatedHPA` and workloads(`Deployment`, `StatefulSet` or other resources that have a subresource named `scale`).
+1. Propose the implementation ideas for involved components, including `karmada-controller-manager`.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories(Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+As a cluster administrator, I know that there will be a sudden peak traffic every days at 9 o'clock AM. So I want to scale up the related service in advance(like 30 minutes) to meet the sharp load peaks, ensure the service is available when the time comes. Also, after the sharp load peaks have gone(like 2 hours latter), I want to scale down the related service to save cloud costs.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go into as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate them? 
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+`CronFederatedHPA` could scale FederatedHPA, but also scale the workloads directly. 
+So if you can set `CronFederatedHPA` to scale the workloads, which also will be scaled by with `FederatedHPA`, it will have conflicts as follows:  
+<image src="./statics/cronfederatedhpa-constraints.png" width="300px">
+
+## Design Details
+### Architecture
+<image src="./statics/cronfederatedhpa-architecture.png" width="300px">
+
+From the above architecture diagram, CronFederatedHPA should do following things:
+* `CronFederatedHPA` allow users to scale the workloads at specific time.
+* `CronFederatedHPA` allow users to set the `FederatedHPA` at specific time.
+
+### API Definition
+In order to satisfy the above requirements, we defined the API as following:
+```go
+type CronFederatedHPA struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec is the specification of the CronFederatedHPASpec.
+	// +required
+	Sepc CronFederatedHPASpec `json:"spec"`
```

Comment:
The proposal doc has "Sepc". This might propagate to the real code during copy-pasting. So beware!

Reply:
Nice finding!